src/flb_io.c

```diff
@@ -118,6 +124,42 @@ int flb_io_net_connect(struct flb_upstream_conn *u_conn,
     return 0;
 }
 
+/*
+ * Perform blocking IO wait
+ * Timeout is provided in milliseconds
+ */
+static flb_io_wait_ret flb_io_wait_sync(struct flb_upstream_conn *u_conn, uint32_t mask,
```

Comment:
I think this function should be part of network subsystem so it can be used by `net_connect_sync`